### PR TITLE
Gui without cdsp instance

### LINKED ISSUE #13: Make GUI work without running CamillaDSP process
As mentioned in https://github.com/HEnquist/camillagui-backend/pull/9#issuecomment-762517607, I would like to make the GUI work without a running CamillaDSP process.

Currently, a running CDSP process is needed to:
- save the current config (via the "Apply" button)
- Load config from a local file
- display the result of the config check (in the "Config status" field)

I see 3 solutions to make this work, if there is no running CDSP process:
1. Simply ignore the failing config check, load and save the config unchecked.
    This sucks for usability, but is very easy to implement.
2. Add a CDSP path to the config and automatically start a backup instance in wait mode, listening on a free port. Always use the backup CDSP, if necessary.
    This would be relatively easy to implement, but can theoretically lead to conflicts, when another program wants the port afterwards.
3. Add a CDSP path to the config and use this CDSP only for config checking.
    A save would be canceled, if the config is invalid.
    The load would be canceled, if the config is invalid. Currently, the running CDSP instance also fills all the optional settings with its defaults. We could either change CDSP's CLI to do it, or simply use the defaults of the GUI. Using defaults of the GUI could theoretically lead to configs breaking, if defaults of the GUI do not match CDSP's defaults.

I prefer solution 3, but if you want the defaults to be supplied by CDSP, I would leave the implementation of the CLI changes to you.

Do you see another way of doing this?
Which solution do you prefer?

### PR DESCRIPTION
**Pull request by JWahle**
Made GUI work without running CDSP instance, specifically:
- load/save working_config
- load config from a local file
- config status

This Fixes #13.
Only the last commit is relevant to this pull request - other commits are from #14.

### CODE REVIEW

File: settings.py [hunk 1]
```diff
@@ -1,21 +1,38 @@
...
\ No newline at end of file
+
+def absolute_path_or_none_if_empty(path):
+    if path:
+        return os.path.abspath(os.path.expanduser(path))
+    else:
+        return None
+
+
+def start_backup_cdsp(config):
```
Comment (HEnquist) [inline settings.py:30]: I would move this function to a new file, "offline.py"

File: settings.py [hunk 2]
```diff
@@ -1,21 +1,38 @@
...
+    backup_cdsp_path = config["backup_camilla_path"]
+    backup_cdsp_port = config["backup_camilla_port"]
+    if backup_cdsp_path and backup_cdsp_port:
+        subprocess.Popen([backup_cdsp_path, "-p", str(backup_cdsp_port), "-w"])
+
+
+config = get_config(config_path)
+start_backup_cdsp(config)
```
Comment (HEnquist) [inline settings.py:38]: this should be done from main

File: views.py [hunk 3]
```diff
@@ -187,10 +197,32 @@ async def set_config(request):
     return web.Response(text="OK")
 
 
+def backup_cdsp(request):
```
Comment (HEnquist) [inline views.py:200]: please move to offline.py

File: views.py [hunk 4]
```diff
@@ -187,10 +197,32 @@ async def set_config(request):
...
+            backup.connect()
+        except ConnectionRefusedError as e:
+            print(e)
+    if backup.is_connected():
+        return backup
+    else:
+        return None
+
+
+def cdsp_or_backup_cdsp(request):
```
Comment (HEnquist) [inline views.py:213]: please move to offline.py

**Comment by HEnquist** [comment]
Looks good. I would move the offline-specific bits to a separate file, and not start any subprocesses from settings.py (can be quite misleading!).

**Comment by JWahle** [comment]
You're right, that's conceptually cleaner. I changed it.

**Pull request merged**